### ISSUE TITLE
MLE 관점에서 본 Cross Entropy

### ISSUE THREAD

**Issue by vmtmxmf5** [vmtmxmf5/Pytorch-#6]
```ruby
CE 최소화 == 로그 가능도 최대화
```
 교차 엔트로피?
Y분포를 사용해서 Y_hat의 entropy를 측정하는 방법.
- -P(x)logQ(x)
엔트로피 속성이 살아 있어서 교차 엔트로피도 작을 수록 좋다.

로그 가능도에 교차 엔트로피를 대입하면, 음의 로그 가능도가 나온다.

즉, 음의 로그 가능도는 작을 수록 교차 엔트로피도 최소화가 되는데, 그 말은 곧 로그 가능도가 최대화가 된다는 의미이다(부호가 반대이므로)

**Comment by vmtmxmf5** [comment]
우리가 추정하는 조건부 확률분포
```math
-log(Pr[y | f_theta(x) ])
```
- 만약 고정된 값을 출력하고 싶다면?
조건부 확률이 따르는 분포의 평균을 출력하면 된다

- 고정된 입력 / 고정된 출력을 안 쓰는 이유는?
다양한 출력을 쓴다는 의미는 곧 분포에서 y_new를 샘플링할 수 있다는 의미다. 예컨대, 강아지 사진을 고정된 1장만 출력하는 게 아니라 다양한 강아지 사진을 생성할 수 있다는 의미이다.

- **조건부 확률분포가 베르누이 분포를 따른다고 가정하면?**

크로스 엔트로피와 같다!!!
```ruby
f_theta(x) = p

p(y | p) = p^y * (1-p)^(1-y)
-log(p(y | p)) = -[ylogp + (1-y)log(1-p)]
``` 
즉, **이산 문제(분류 문제)에서는 크로스 엔트로피를 쓰는게 MLE**임을 알 수 있다!

또, 엔트로피를 가장 높일 수 있다! (가장 등확률에 가까우므로 공정함)


P.s. 위와 같은 수식으로 -log(Pr[y | f_theta(x) ])이 가우시안 분포를 따른다고 가정하면, **MSE**와 비례함을 보일 수 있다! 즉, 연속형 자료의 경우 MSE가 MLE인 셈이다.

**Comment by vmtmxmf5** [comment]
https://hyunlee103.tistory.com/12  

https://ratsgo.github.io/deep%20learning/2017/09/24/loss/  

https://ratsgo.github.io/statistics/2017/09/22/information/
